lib/Sema/CSDiag.cpp

```diff
@@ -6508,7 +6510,69 @@ bool FailureDiagnosis::visitClosureExpr(ClosureExpr *CE) {
         }
         return true;
       }
-      
+
+      if (inferredArgCount == 1 && actualArgCount > 1) {
+        // Let's see if inferred argument is actually a tuple inside of Paren.
+        if (auto *argTupleTy = inferredArgType->getAs<TupleType>()) {
+          // Looks like the number of closure parameters matches number
+          // of inferred arguments, which means we can warn about SE-0110
```

Comment:
I would avoid referencing the SE proposal like this because it results in an extra step to go figure out what you're talking about. You could just say "...we can emit an error about an attempt to make use of tuple splat or tuple destructuring..."